docs/DeveloperGuide.md

```diff
@@ -4,30 +4,148 @@
+`Parser#extractCommandKeyword(userInput)` to parse the input and determine that the `delete` command is called.
+
+Step 2. `Duke` instantiates a new `CommandDelete` and calls `CommandDelete#execute()`, which then calls
+`expenseList.getExpenseList()` and `parser.extractIndex(input)`.
+
+Step 3. `CommandDelete#execute()` removes the expense at index specified by the user.
+
+### Monthly Overview
+
+This mechanism is facilitated by `CommandOverview`, which extends `Command`. It makes use of output from `Parser`
+to extract `month` and `year` from user input. It then calls on `Monthly Overview` if both `month` and 'year' 
+are not null, which makes use of `MonthFilter` to filter out expenses in that specific month and returns sum by 
+category sorted in descending order before printing out the final overview in the intended format.
+
+Given below is the sequence diagram to explain how the 'monthly overview' mechanism behaves.
+![](diagrams/MonthlyOverview.png)
+
+
+### 'Total' feature
+This mechanism is facilitated by `CommandTotal`.
+
+`CommandTotal` implements the following operations:
+- `CommandTotal#calculateTotal()` -- Iterates through the given expense list and calculates the total expenses from its amount and currency exchange rate.
+- `CommandTotal#getTotal()` -- Returns the total in 2 decimal places.
+
+
+Displayed below is a part of the class diagram for `CommandTotal`.
+![](diagrams/TotalFeature.png)
+Give below is an example usage of the feature.
+
+Step 1. The user executes `total`. Duke calls on `CommandTotal#execute()` to calculate the total expenses, which in turn calls
+`expenseList#getExpenseList()` to retrieve the expense list.
+
+Step 2. The expense list is iterated through and `total` is obtained by summing the product of each expenseAmount and rate.
 
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+- has a need to manage daily expenses
+- travels overseas often and want to keep track of their expenses throughout their travels in different currencies
+- prefer desktop apps over other types
+- can type fast
+- prefers typing to mouse interactions
+- is reasonably comfortable using CLI apps
 
 ### Value proposition
 
 {Describe the value proposition: what problem does it solve?}
+- Our expense tracker is a simple program that allows users to track their expenses
+- With simple commands and ease of use, users are able to use our tracker with little to no experience.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...          | I want to ...                                               | So that I can ...                                                                        |
```

Comment:
Possible formatting error in the table, there is an extra blank column to the right of "So that I can..."

note that it is not necessary to match the whitespaces one-to-one for the table in markdown, as github will automatically format it 

ie. you can do this:

| v1.0 | new user | add a new entry for my  | - |
| v1.0 | user | delete specific expenses | - |
| v1.0 | user | add expenses with dates | track how much I spend each day |
| v1.0 | user| add expenses of different categories | keep track of what I spend on |

and github will format this automatically into a nice table
![table](https://user-images.githubusercontent.com/88242862/228703089-84ef96e1-7708-44b9-b51b-3e5a193d7f91.png)